Index: index.php
===================================================================
RCS file: /cvs/drupal/drupal/index.php,v
retrieving revision 1.99
diff -u -p -r1.99 index.php
--- index.php	15 Oct 2009 14:07:25 -0000	1.99
+++ index.php	3 Feb 2011 18:35:08 -0000
@@ -11,7 +11,19 @@
  * All Drupal code is released under the GNU General Public License.
  * See COPYRIGHT.txt and LICENSE.txt.
  */
-
+##GardensExcludeFromExportStart################################################
+// When a Gardens site is deleted and a new site is created using the same URL
+// (and therefore with the same symlink in the sites directory in the file
+// system, but with the symlink pointed to a different target), some
+// combination of the PHP realpath cache and the APC cache can result in the
+// old site's settings.php and other files still being used when the new site
+// is visited. To fix this, clearing out the PHP static cache at the beginning
+// of all Drupal page requests seems to be the only reliable solution. This
+// means that in Drupal Gardens, we do not have any cross-request caching of
+// files in PHP, but still can take advantage of the cache within a single page
+// request.
+clearstatcache();
+##################################################GardensExcludeFromExportEnd##
 /**
  * Root directory of Drupal installation.
  */
