diff --git a/coppa_lite.module b/coppa_lite.module
index e70dbeb..9c08443 100755
--- a/coppa_lite.module
+++ b/coppa_lite.module
@@ -251,7 +251,9 @@ function coppa_lite_user_validate($form, &$form_state) {
       break;
 
     case COPPA_LITE_COOKIE_PER_BROWSER:
+        
       $cookie_index = md5("coppa_lite");
+      $deny_all_registration =  $cookie_index;
       break;
   }
 
@@ -271,6 +273,9 @@ function coppa_lite_user_validate($form, &$form_state) {
     // Set Cookie age.
     $expire = time() + (60 * 60 * 24 * variable_get('coppa_lite_cookie_expire', 1));
     setcookie($cookie_index, date("G:i - m/d/y"), $expire, '/');
+    if(isset($deny_all_registration)){
+        setcookie('coppa_deny_all', $deny_all_registration, $expire, '/');
+    }
     form_set_error('', check_plain($warning_message));
   }
 }
