diff --git a/docroot/sites/all/modules/contrib/honeypot/honeypot.admin.inc b/docroot/sites/all/modules/contrib/honeypot/honeypot.admin.inc
index 37bcba7..e975482 100644
--- a/docroot/sites/all/modules/contrib/honeypot/honeypot.admin.inc
+++ b/docroot/sites/all/modules/contrib/honeypot/honeypot.admin.inc
@@ -37,6 +37,13 @@ function honeypot_admin_form($form, &$form_state) {
     '#required' => TRUE,
     '#size' => 30,
   );
+  $form['configuration']['honeypot_element_label'] = array(
+        '#type' => 'textfield',
+        '#title' => t('Honeypot element form label'),
+        '#default_value' => variable_get('honeypot_element_label', 'Leave this field blank'),
+        '#size' => 255,
+        '#description' => t("Label for front end - Default is 'Leave this field blank'"),
+      );
   $form['configuration']['honeypot_time_limit'] = array(
     '#type' => 'textfield',
     '#title' => t('Honeypot time limit'),
diff --git a/docroot/sites/all/modules/contrib/honeypot/honeypot.module b/docroot/sites/all/modules/contrib/honeypot/honeypot.module
index 22d3de1..c1c4a48 100644
--- a/docroot/sites/all/modules/contrib/honeypot/honeypot.module
+++ b/docroot/sites/all/modules/contrib/honeypot/honeypot.module
@@ -184,12 +184,13 @@ function honeypot_add_form_protection(&$form, &$form_state, $options = array())
   if (in_array('honeypot', $options)) {
     // Get the element name (default is generic 'url').
     $honeypot_element = variable_get('honeypot_element_name', 'url');
+    $honeypot_label = variable_get('honeypot_element_label', t('Leave this field blank'));
 
     // Build the honeypot element.
     $honeypot_class = $honeypot_element . '-textfield';
-    $form[$honeypot_element] = array(
+   $form[$honeypot_element] = array(
       '#type' => 'textfield',
-      '#title' => t('Leave this field blank'),
+      '#title' => $honeypot_label,
       '#size' => 20,
       '#weight' => 100,
       '#attributes' => array('autocomplete' => 'off'),
