diff --git a/field_group.js b/field_group.js
index 0b0a565..df5be3b 100644
--- a/field_group.js
+++ b/field_group.js
@@ -50,7 +50,8 @@ Drupal.FieldGroup.Effects.processAccordion = {
       });
 
       wrapper.accordion({
-        heightStyle: "content",
+        heightStyle: "content", // jQuery UI >= v1.9
+        autoHeight: false,      // jQuery UI < v1.9
         active: active_index,
         collapsible: true,
         changestart: function(event, ui) {
