diff --git a/password_policy.admin.inc b/password_policy.admin.inc
index 733b17e..3c3f2e6 100644
--- a/password_policy.admin.inc
+++ b/password_policy.admin.inc
@@ -466,6 +466,15 @@ function password_policy_admin_form_validate($form, &$form_state) {
   if (!empty($form_state['values']['warning']) && !preg_match("/^[0-9,]+$/", $form_state['values']['warning'])) {
     form_set_error('warning', t('Warning must only contain digits and commas.'));
   }
+  if(!isset($form_state['values']['pid']) && $form_state['values']['op'] != t('Delete')) {
+    $query = db_select("password_policy", "p");
+    $query->condition("p.name", $form_state['values']['name']);
+    $query->fields("p", array("pid"));
+    $results = $query->execute()->rowCount();
+    if($results > 0) {
+      form_set_error('name', t('A password policy with same name already exists. Please choose a different name'));
+    }
+  }
 }
 
 /**
